scripts/lspArtifact.ts

```diff
@@ -0,0 +1,181 @@
+import * as https from 'https'
+import * as fs from 'fs'
+import * as crypto from 'crypto'
+import * as path from 'path'
+import * as os from 'os'
+import AdmZip from 'adm-zip'
+
+interface ManifestContent {
+    filename: string
+    url: string
+    hashes: string[]
+    bytes: number
+}
+
+interface ManifestTarget {
+    platform: string
+    arch: string
+    contents: ManifestContent[]
+}
+
+interface ManifestVersion {
+    serverVersion: string
+    isDelisted: boolean
+    targets: ManifestTarget[]
+}
+
+interface Manifest {
+    versions: ManifestVersion[]
+}
+async function verifyFileHash(filePath: string, expectedHash: string): Promise<boolean> {
+    return new Promise((resolve, reject) => {
+        const hash = crypto.createHash('sha384')
+        const stream = fs.createReadStream(filePath)
+
+        stream.on('data', (data) => {
+            hash.update(data)
+        })
+
+        stream.on('end', () => {
+            const fileHash = hash.digest('hex')
+            // Remove 'sha384:' prefix from expected hash if present
+            const expectedHashValue = expectedHash.replace('sha384:', '')
+            resolve(fileHash === expectedHashValue)
+        })
+
+        stream.on('error', reject)
+    })
+}
+
+async function ensureDirectoryExists(dirPath: string): Promise<void> {
+    if (!fs.existsSync(dirPath)) {
+        await fs.promises.mkdir(dirPath, { recursive: true })
+    }
+}
+
+export async function downloadLanguageServer(): Promise<void> {
+    const tempDir = path.join(os.tmpdir(), 'amazonq-download-temp')
+    const resourcesDir = path.join(__dirname, '../packages/amazonq/resources/language-server')
+
+    // clear previous cached language server
+    try {
+        if (fs.existsSync(resourcesDir)) {
+            fs.rmdirSync(resourcesDir, { recursive: true })
+        }
+    } catch (e) {
+        throw Error(`Failed to clean up language server ${resourcesDir}`)
+    }
+
+    await ensureDirectoryExists(tempDir)
+    await ensureDirectoryExists(resourcesDir)
+
+    return new Promise((resolve, reject) => {
+        const manifestUrl = 'https://aws-toolkit-language-servers.amazonaws.com/qAgenticChatServer/0/manifest.json'
```

Comment:
@leigaol Future note: this could be a source of confusion for cases like https://github.com/aws/aws-toolkit-vscode/commit/756c286c8728b5c5434971ecd58a96a1c56a992b when an alpha manifest is used. Unless this location is also updated.